spyder/widgets/editor.py

```diff
@@ -1302,7 +1304,11 @@ def file_saved_in_other_editorstack(self, index, filename):
     def select_savename(self, original_filename):
         self.redirect_stdio.emit(False)
         filename, _selfilter = getsavefilename(self, _("Save file"),
-                                               original_filename)
+                                               original_filename,
+                                               get_edit_filters(),
+                                               get_filter(get_edit_filetypes(),

```

Comment:
Perhaps if we change here `get_edit_filetypes` to `EDIT_FILETYPES` we could get the expected result :-)